edx/edx-enterprise
# ENT-100: Add EnterpriseCourseEnrollment object

**Description:**

This pull request adds an EnterpriseCourseEnrollment object. At present, this object links with an EnterpriseCustomerUser and a course ID to denote a course in which a user has been registered with the sponsorship of an EnterpriseCustomer. It also holds a nullable Boolean field for data sharing consent state.

In addition, this pull request adds logic to automatically create an EnterpriseCourseEnrollment object when manually enrolling learners from the Manage Learners admin, and will create one when converting a PendingEnrollment to a full enrollment.

**JIRA:** [ENT-100](https://openedx.atlassian.net/browse/ENT-100)

**Dependencies:** None

**Merge deadline:** 30 January 2017

**Testing instructions:**

1. Using the Manage Learners view, link a learner with an existing account on edx-platform and simultaneously enroll them in a course.
    - In the Django shell:
    ```python
        from enterprise.models import EnterpriseCourseEnrollment as ECE
        ECE.objects.all()
    ```
    - Observe that there is now an EnterpriseCourseEnrollment for the user.
2. Using the Manage Learners view, link a learner by email address who does not have an existing account on edx-platform, simultaneously enrolling them in a course.
3. Create an account on edx-platform, using the email address you just linked a pending learner with.
    - In the Django shell:
    ```python
        from enterprise.models import EnterpriseCourseEnrollment as ECE
        ECE.objects.all()
    ```
    - Observe that there are now two EnterpriseCourseEnrollment objects; one for the learner you previously created; one for the learner you just now linked.

**Reviewers:**

**Merge checklist:**

**Post merge:**
      finished.
